pkg/cloud/services/ec2/instances.go

```diff
@@ -564,6 +573,23 @@ func (s *Service) getImageRootDevice(imageID string) (*string, error) {
 	return output.Images[0].RootDeviceName, nil
 }
 
+func (s *Service) getImageSnapshotSize(imageID string) (*int64, error) {
+	input := &ec2.DescribeImagesInput{
+		ImageIds: []*string{aws.String(imageID)},
+	}
+
+	output, err := s.scope.EC2.DescribeImages(input)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(output.Images) == 0 {
+		return nil, errors.Errorf("no images returned when looking up ID %q", imageID)
+	}
+
+	return output.Images[0].BlockDeviceMappings[0].Ebs.VolumeSize, nil
+}
+
```

Comment:
This func is basically identical to the one above (to get the root device name), just returning a different attribute. Thoughts on adjusting the above one (which is only used once) to return size and name to save an AWS call?

Reply:
+1 to consolidating the calls. This can also be handled as a followup.